core/state_transition.go

```diff
@@ -74,6 +76,8 @@ type Message interface {
 	Nonce() uint64
 	CheckNonce() bool
 	Data() []byte
+
+	TxType() types.TransactionType
```

Comment:
so we are actually not adding new field into transaction struct itself?

Reply:
yes